src/w3iProxy/pushProviders/internalPushProvider.ts

```diff
@@ -134,8 +134,19 @@ export default class InternalPushProvider implements W3iPushProvider {
 
     // Retrieving FCM token needs to be client side, outside the service worker.
     const token = await getToken(messaging, {
-      vapidKey:
-        'BCnI0mkpH3LvHRF-dREPCdvBFk24oveWy4JBuINzWcu8JXhmCDkczDmHM9RubzsQrv60UKFk-MKozVjRRzvx1X4'
+      vapidKey: import.meta.env.VITE_VAPID_KEY
+    })
+
+    this.pushClient.once('push_subscription', subEv => {
```

Comment:
Nicer with the event based approach but seems to be missing this check now? https://github.com/WalletConnect/web3inbox/pull/187/files/b55cec45c4255476666e9f619f44c3e64d69ae67..c58a1e770e9e4bc794db9a9cb9499944343e1e85#diff-6bd271dbba7eb11f9fa29007d2477bfedc6f57605e5ae00ca8cfa6eb75d9dec0L158

Reply:
Why would we need this check now? Attaching this listener right before firing the subscribe is fired, surely it'll be related? 

The only edge case i can think of is if someone opened the explorer and spammed clicking subscribe on multiple subscriptions. I'll add the guard